producer/src/contract/contract.go

```diff
@@ -65,38 +64,108 @@ the block index containing the contract of the claimed yield,
 the index of the claimed yield in the contract,
 and the public key of the claimant
 */
-type Claim struct{}
+type Claim struct {
+	PreviousContractHash []byte
+	BlockIndex           uint64
+	YieldIndex           uint16
+	PublicKey            ecdsa.PublicKey
+}
 
-/*
-Should scan the Unclaimed Yield Pool for a yield
+func MakeClaim(database string, claimant ecdsa.PublicKey, value uint64) (Claim, error) {
+	claimantHexHash := hex.EncodeToString(block.HashSHA256(keys.EncodePublicKey(&claimant)))
+	//open datase first
+	dbConn, err := sql.Open("sqlite3", database)
+	if err != nil {
+		log.Printf("Error! unable to open databse: %v\n", err)
+		return Claim{}, err
+	}
+	defer dbConn.Close()
 
-Prioritize yields that are closest to the value parameter,
-ie. MIN(abs(value - yieldValue0), abs(value - yieldValue1), ... abs(value - yieldValueN))
+	//first query the db for any yeilds equal or above
+	sqlQuery := `SELECT * FROM uy WHERE value >= $1 AND holder = $2 ORDER BY value ASC LIMIT 1;`
+	rows, err2 := dbConn.Query(sqlQuery, value, claimantHexHash)
+	if err2 != nil {
+		log.Printf("Error! Unable to execte query: %v\n", err2)
+	}
 
-Case 1:
-If the claimed yield is less than or equal to the value, return
-the claim and a nil for the error
+	var dbHeight uint64
+	var dbContract string
+	var dbIndex uint16
+	var dbHolder string
+	var dbValue uint64
+
+	if rows.Next() {
+		if err3 := rows.Scan(&dbHeight, &dbContract, &dbIndex, &dbHolder, &dbValue); err3 != nil {
+			log.Fatal(err3)
+		}
+		c := Claim{}
+		var rtnErr error //nil by default
+		//determine if there is change
+		if dbValue > value {
+			rtnErr = ChangeError{Change: dbValue - value}
+		}
+		c.PreviousContractHash, _ = hex.DecodeString(dbContract)
+		c.BlockIndex = dbHeight
+		c.YieldIndex = dbIndex
+		c.PublicKey = claimant
+
+		rows.Close()
+		return c, rtnErr
+	}
 
-Case 2:
-If the claimed yield exceeds the value, return
-the claim as usual and a custom error struct with
-the difference as a uint64 field called "Change"
+	//else we need to select a yield that has less value
+	sqlQuery = `SELECT * FROM uy WHERE holder = $1 ORDER BY value DESC LIMIT 1;`
+	rows2, err3 := dbConn.Query(sqlQuery, claimantHexHash)
```

Comment:
you dont need to create multiple `rows` and `errs`, you can simply reassign these with a equal sign without a `:`

Reply:
You right